components/SwitchNetwork/SwitchNetwork.tsx

```diff
@@ -0,0 +1,94 @@
+import { ethers, BigNumber } from 'ethers';
+import styled from 'styled-components';
+import { useState, useCallback, useEffect } from 'react';
+import { OPTIMISM_NETWORKS } from '@synthetixio/optimism-networks';
+
+import Connector from 'containers/Connector';
+
+import Button from 'components/Button';
+import BaseModal from 'components/BaseModal';
+
+import { nameToIdMapping, chainIdMapping } from 'constants/networks';
+
+const SwitchNetwork = ({ pool, poolNetwork, isReady }: any) => {
+	const { network, walletAddress, provider } = Connector.useContainer();
+	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+
+	const poolNetworkId = nameToIdMapping[poolNetwork ?? '']?.id ?? '';
+
+	useEffect(() => {
```

Comment:
You can probably use `useMemo` here

Reply:
We need to set a new state, I think `useEffect` fits better for this.